cloud/services/compute/machineimages/reconcile.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2024 The Forge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machineimages
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/compute/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/forge-build/forge-provider-gcp/cloud/gcperrors"
+)
+
+// Reconcile ensures that an image is created from an instance.
+func (s *Service) Reconcile(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	if !s.scope.IsProvisionerReady() {
```

Comment:
Here you can check isMachineReady and IsProvisionerReady

Reply:
If isProviionerReady means that the Machine is Ready